lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -89,104 +89,242 @@ export class BrowserStorage extends CacheManager {
+        const appMetadata: AppMetadataEntity = CacheManager.toObject(new AppMetadataEntity(), JSON.parse(value));
+        if (AppMetadataEntity.isAppMetadataEntity(appMetadataKey, appMetadata)) {
+            return appMetadata;
+        }
+        return null;
+
+    }
+
+    /**
+     * set appMetadata entity to the platform cache
+     * @param appMetadata
+     */
+    setAppMetadata(appMetadata: AppMetadataEntity): void {
+        const appMetadataKey = appMetadata.generateAppMetadataKey();
+        this.setItem(appMetadataKey, JSON.stringify(appMetadata));
+    }
+
+    /**
+     * fetch server telemetry entity from the platform cache
+     * @param serverTelemetryKey
+     */
+    getServerTelemetry(serverTelemetryKey: string): ServerTelemetryEntity | null {
+        const value = this.getItem(serverTelemetryKey);
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+        const serverTelemetryEntity = CacheManager.toObject(new ServerTelemetryEntity(), JSON.parse(value));
+        if (ServerTelemetryEntity.isServerTelemetryEntity(serverTelemetryKey, serverTelemetryEntity)) {
+            return serverTelemetryEntity;
+        }
+        return null;
+    }
+
+    /**
+     * set server telemetry entity to the platform cache
+     * @param serverTelemetryKey
+     * @param serverTelemetry
+     */
+    setServerTelemetry(serverTelemetryKey: string, serverTelemetry: ServerTelemetryEntity): void {
+        this.setItem(serverTelemetryKey, JSON.stringify(serverTelemetry));
+    }
+
+    /**
+     * fetch throttling entity from the platform cache
+     * @param throttlingCacheKey
+     */
+    getThrottlingCache(throttlingCacheKey: string): ThrottlingEntity | null {
+        const value = this.getItem(throttlingCacheKey);
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+        const throttlingCache = CacheManager.toObject(new ThrottlingEntity(), JSON.parse(value));
+
+        if (ThrottlingEntity.isThrottlingEntity(throttlingCacheKey, throttlingCache)) {
+            return throttlingCache;
+        }
+        return null;
+    }
+
+    /**
+     * set throttling entity to the platform cache
+     * @param throttlingCacheKey
+     * @param throttlingCache
+     */
+    setThrottlingCache(throttlingCacheKey: string, throttlingCache: ThrottlingEntity): void {
+        this.setItem(throttlingCacheKey, JSON.stringify(throttlingCache));
+    }
+
     /**
      * Gets cache item with given key.
      * Will retrieve frm cookies if storeAuthStateInCookie is set to true.
      * @param key
      */
-    getItem(key: string, type: string): string | object {
-        const value = this.windowStorage.getItem(key);
+    getTemporaryCache(cacheKey: string, generateKey?: boolean): string {
+        const key = generateKey ? this.generateCacheKey(cacheKey) : cacheKey;
+
+        const itemCookie = this.getItemCookie(key);
+        if (this.cacheConfig.storeAuthStateInCookie && itemCookie) {
```

Comment:
Per #2328, we need to only attempt to read cookie value if `storeAuthStateInCookie` is true.

Reply:
Applied your suggestion